src/reconcile/rec_row.c

```diff
@@ -460,11 +464,17 @@ __wti_rec_row_int(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
 
         /* Update compression state. */
         __rec_key_state_update(r, false);
+
+        /*
+         * Set the ref_changes state to zero if there were no concurrent changes while reconciling
+         * the internal page.
+         */
+        __wt_atomic_casv16(&ref->ref_changes, prev_ref_changes, 0);
     }
     WT_INTL_FOREACH_END;
 
     /* Write the remnant page. */
-    return (__wti_rec_split_finish(session, r));
```

Comment:
I don't quite understand this change - is it a bugfix where we weren't releasing the hazard pointer previously?

Reply:
This change is wrong :)

Reply:
Yes, it's a mistake, I'll change it back.